IzuOvando/ticket_printer_web_app
# 🖨️ Feature/printers-connection

## What whats done ❓
- ✨Added functionality of printers which:
  - ➕ Can be added
  - ✏️ Can be edited
  - 🗑️ Can be deleted
  - 🔄️ Can be reconnected
  - 🖨️ Can be tested
 Also when exit the app and going back there it is an auto-reconnection to check their status (`online `or `offline`)

All of that with an amazing UI/UX ✨

> [!NOTE]
> The only thing that needs to be done is handle success/failure print and reconnect before seen printers and idle

## Type of change ♻️

- [ ] realese
- [ ] support

> [!WARNING]
> There are many more dependecies like the state manager `zustand`, so install dependencies again.

## How can be tested? 🔍

Login, go to printers and try to connect many printers (one of them should be connected)

## Screenshots 🖼️ 

![image](https://github.com/IzuOvando/ticket_printer_web_app/assets/30848819/bd073a42-966d-4660-b4dc-ec2b6c505e03)
![image](https://github.com/IzuOvando/ticket_printer_web_app/assets/30848819/c16e67da-da0b-43b9-94c7-83b649bec951)
![image](https://github.com/IzuOvando/ticket_printer_web_app/assets/30848819/e930b234-c410-4e41-93c4-5af27c87f01b)
![image](https://github.com/IzuOvando/ticket_printer_web_app/assets/30848819/497f1c04-b293-4eff-be44-83830db5b7b7)
![image](https://github.com/IzuOvando/ticket_printer_web_app/assets/30848819/9f9d22c7-7903-43a9-84ed-3bf5a038ded3)
